src/ddscxx/include/org/eclipse/cyclonedds/topic/cdr_dummys.hpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright(c) 2021 ADLINK Technology Limited and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+ * v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+#ifndef CDR_DUMMYS_HPP_
+#define CDR_DUMMYS_HPP_
+
+#include <string>
+#include <vector>
+#include <array>
+
```

Comment:
Maybe the choice of the prefix idl_ is not the ideal one, could be that something like cdr_ is better?

Reply:
Indeed, they should not be prefixed with `idl_`. Because 1) These types have nothing to do with IDL, 2) They are not in a reserved namespace and 3) Using a prefix as a namespace is the C way of doings because namespaces aren't available there. I think `org::eclipse::cyclonedds::<something>` or `dds::<something>` should be used (probably the former). There's `dds/core/types.hpp`, so perhaps `org/eclipse/cyclonedds/core/types.hpp` is the right location? They're mainly "aliases" to make serialization more straightforward right? I'd probably opt for `org::eclipse::cyclonedds::core::string` etc. I'm a little worried what happens if a user provides a type that does not match the requirements though. Will the serializer/compiler handle that properly?